crates/sui-types/src/crypto.rs

```diff
@@ -74,17 +74,28 @@ pub const DERVIATION_PATH_PURPOSE_ED25519: u32 = 44;
 pub const DERVIATION_PATH_PURPOSE_SECP256K1: u32 = 54;
 pub const TBLS_RANDOMNESS_OBJECT_DOMAIN: &[u8; 10] = b"randomness";
 
-// Creates a proof that the keypair is possesed, as well as binds this proof to a specific SuiAddress.
-pub fn generate_proof_of_possession<K: KeypairTraits>(
-    keypair: &K,
+/// Creates a proof of that the authority account address is owned by the
+/// holder of authority protocol key, and also ensures that the authority
+/// protocol public key exists. A proof of possession is an authority
+/// signature committed over the intent message `intent || message` (See
+/// more at [struct IntentMessage] and [struct Intent]) where the message is
+/// constructed as `kosk domain || authority_pubkey_bytes || authority_account_address`.
+pub fn generate_proof_of_possession(
+    keypair: &AuthorityKeyPair,
     address: SuiAddress,
-) -> <K as KeypairTraits>::Sig {
+) -> AuthoritySignature {
     let mut domain_with_pk: Vec<u8> = Vec::new();
     domain_with_pk.extend_from_slice(PROOF_OF_POSSESSION_DOMAIN);
     domain_with_pk.extend_from_slice(keypair.public().as_bytes());
     domain_with_pk.extend_from_slice(address.as_ref());
-    // TODO (joyqvq): Use Signature::new_secure
-    keypair.sign(&domain_with_pk[..])
+    AuthoritySignature::new_secure(
+        &IntentMessage::new(
+            Intent::default().with_scope(IntentScope::ProofOfPossession),
+            domain_with_pk,
+        ),
+        &0,
```

Comment:
this verify_secure comes from the generic impl of an authority signature where epoch_id is always required. to keep the interface minimal change, we can always commit to epoch = 0. wdyt? alternatively, we can remove this field from pop signing, but we will need a different set of API for verify_secure

Reply:
that's probably a good idea to have the pop commit to the epoch since we do know what next epoch the validator request to update metadata for protocol pubkey. or in genesis, epoch = 0. 

would this be any issue? @longbowlu

Reply:
it's fine at this stage to add as last element, as we did for authority sigs as well. We could add a TODO to explore later.

Reply:
if there's no requirement to rotate pop every X epoch, I'm not sure what's the benefit of including epoch into the signature? Also it could be a bit tricky to do in our codebase today as there are active validators, pending validators and candidate validators, the "epoch" they commit to may mean differently.

Reply:
so for a validator dont know what epoch will there pubkey/pop will be accepted and validated?